src/Storages/ExternalDataSourceConfiguration.cpp

```diff
@@ -159,17 +165,31 @@ std::optional<ExternalDataSourceConfig> getExternalDataSourceConfiguration(const
     return std::nullopt;
 }
 
+static void validateConfigKeys(
+    const Poco::Util::AbstractConfiguration & dict_config, const String & config_prefix, const std::unordered_set<std::string_view> & allowed_keys)
+{
+    Poco::Util::AbstractConfiguration::Keys config_keys;
+    dict_config.keys(config_prefix, config_keys);
+    for (const auto & config_key : config_keys)
+    {
+        if (allowed_keys.contains(config_key) || config_key.starts_with("replica"))
+            continue;
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unexpected key `{}` in dictionary source configuration", config_key);
```

Comment:
Maybe make this check optional (setting enabled to default)? I'm not sure about ALL possible dictionaries sources....

Reply:
No I think it should either be always enabled or not exist at all.

Reply:
btw this method is not used for ALL dictionaries, just for external dictionaries (almost all) + clickhouse dictionary source.